ClosedXML/Excel/Cells/XLCell.cs

```diff
@@ -2130,31 +2130,55 @@ public IXLRangeAddress FormulaReference
             }
         }
 
-        public IXLRange CurrentRegion
-        {
-            get
-            {
-                return this.Worksheet.Range(FindCurrentRegion(this.AsRange()));
-            }
-        }
+        public IXLRange CurrentRegion => Worksheet.Range(FindCurrentRegion(AsRange()));
 
         internal IXLRangeAddress FindCurrentRegion(IXLRangeBase range)
         {
-            var rangeAddress = range.RangeAddress;
+            var sheet = range.Worksheet;
+            var currentRangeAddress = range.RangeAddress;
 
-            var filledCells = range
-                .SurroundingCells(c => !(c as XLCell).IsEmpty(XLCellsUsedOptions.AllContents))
-                .Concat(this.Worksheet.Range(rangeAddress).Cells());
+            var minRow = currentRangeAddress.FirstAddress.RowNumber;
+            var minCol = currentRangeAddress.FirstAddress.ColumnNumber;
+            var maxRow = currentRangeAddress.LastAddress.RowNumber;
+            var maxCol = currentRangeAddress.LastAddress.ColumnNumber;
 
-            var grownRangeAddress = new XLRangeAddress(
-                new XLAddress(this.Worksheet, filledCells.Min(c => c.Address.RowNumber), filledCells.Min(c => c.Address.ColumnNumber), false, false),
-                new XLAddress(this.Worksheet, filledCells.Max(c => c.Address.RowNumber), filledCells.Max(c => c.Address.ColumnNumber), false, false)
-            );
+            bool hasRegionExpanded;
 
-            if (rangeAddress.Equals(grownRangeAddress))
-                return this.Worksheet.Range(grownRangeAddress).RangeAddress;
-            else
-                return FindCurrentRegion(this.Worksheet.Range(grownRangeAddress));
+            do
+            {
+                hasRegionExpanded = false;
+
+                for (var row = Math.Max(minRow - 1, 1); row <= Math.Min(maxRow + 1, 1048576); row++)
+                {
+                    for (var col = Math.Max(minCol - 1, 1); col <= Math.Min(maxCol + 1, 16384); col++)
+                    {
+                        if (row >= minRow && row <= maxRow && col >= minCol && col <= maxCol)
+                        {
+                            continue;
+                        }
+
+                        var cell = sheet.Cell(row, col);
+
+                        if (cell.IsEmpty(XLCellsUsedOptions.AllContents))
+                            continue;
+
+                        hasRegionExpanded = true;
+
+                        minRow = Math.Min(minRow, row);
+                        minCol = Math.Min(minCol, col);
+                        maxRow = Math.Max(maxRow, row);
+                        maxCol = Math.Max(maxCol, col);
+                    }
+                }
+
+                if (!hasRegionExpanded)
+                    continue;
```

Comment:
I think break is more appropriate at this place than continue. Continue jumps to the loop condition that immediately ends.

The if condition is likely redundant, when the currentRangeAddres is moved out of the loop.